### ISSUE TITLE
arrow keys not working after interacting with an extension

### ISSUE THREAD

**Issue by Pogeres** [zen-browser/desktop#2743]
### Captchas

- [X] I have read the instructions.
- [X] I have searched existing issues and avoided creating duplicates.
- [x] I am not filing an enhancement request.

### What happened?

I was unable to navigate using the arrow keys (up, down, left, right), as well as the Home and End keys, in the browser. These keys don’t function in areas like the address bar, text fields within forms, and any editable text sections on web pages, which disrupts basic navigation. In keyboard testing sites, it shows that the keys are being pressed, so the issue seems isolated to the browser itself. Outside the browser, these keys work as expected. This affects usability by preventing efficient navigation and editing in essential browser areas.

### Reproducible?

- [X] I have checked that this issue cannot be reproduced on Mozilla Firefox.

### Version

1.0.1-a.17

### What platform are you seeing the problem on?

Windows

**Comment by red-crate** [comment]
**I am experiencing the same bug.**

Browser version is **1.0.1-a.19**, and I'm using **Windows 11**.

I recall that before the update, the arrow keys worked well, but currently, not only in the URL bar but also in this text area where I'm typing now, the arrow keys do not function.

I can only move back to the text I just wrote using the mouse cursor; I cannot move the cursor using the arrow keys (→/←/↑/↓) or the Home/End keys.

It appears that arrow keys assigned to shortcuts are working properly.

In case it might help, I'd like to mention that I'm a Korean user, and I wonder if this could be a bug related to the **IME (Input Method Editor)**.

**Comment by Faxendo** [comment]
Same issue here on following config :

**Browser version :** 1.0.1-a.19
**OS :** MacOS Sequoia 15.1
**Keyboard Layout :** AZERTY (French)

**Comment by petersjoo** [comment]
Same issue, arrow keys not working in any form fields
Edit: Just noticed that **tab button** is not working  either, very annoying when filling forms :/
Edit 2: Reboot made arrows and tab work again.
Win 11
1.0.1-a.19
QWERTY (Swedish)

**Comment by TheDavidFactor** [comment]
I just experienced this same issue. Reboot fixed it. Before reboot I tested in both firefox and edge and keys were working normally in those browsers.

Win 11
1.0.2-b.2 (Firefox 133.0.3) (64-bit)
QWERTY (English)

**Comment by MemQu** [comment]
I'm having this issue too.

Win 11
1.0.2-b.2

English (US)

It resolves sometimes after I restart, but comes back again on later starts.

**Comment by MemQu** [comment]
After some testing, I find this happens after clicking on an extension in the top bar. Arrow keys and shift + tab stops working, requiring a restart.

**Comment by rickybrent** [comment]
> After some testing, I find this happens after clicking on an extension in the top bar. Arrow keys and shift + tab stops working, requiring a restart.

That seems to be it for me as well! (Linux, KDE -- and crucially for this bug, multiple toolbars.)

I'm able to use the arrow keys again without a restart by:
1) Switching from Multiple Toolbars to Single Toolbar. (Note: Arrow key input will still be broken after this step.)
2) Click an the extension button again, now located in the Single Toolbar. (Arrow key input is now fixed.)
3) Optionally, switch back to Multiple Toolbars.

**Comment by jonaskohl** [comment]
Exactly the same issue here on macOS 14.7.1 and Zen 1.0.2-b.2.  
Reproducible by clicking on an extension inside the extension menu from the top bar (note this does not apply to extensions pinned to the top bar directly).

**Comment by MemQu** [comment]
> Exactly the same issue here on macOS 14.7.1 and Zen 1.0.2-b.2. Reproducible by clicking on an extension inside the extension menu from the top bar (note this does not apply to extensions pinned to the top bar directly).

It must be different on windows. Both unpinned and pinned extensions affect zen on my windows machine

**Comment by Volzhanka63** [comment]
Same issue, for a week or so. 
Windows 11
1.0.2-b.2 (Firefox 133.0.3) (64-bit)

**Comment by jonaskohl** [comment]
Opening DevTools also triggers the bug

**Comment by joelazar** [comment]
~~It seems to be fixed now in version 1.0.2-b3.~~ Nope, still there.

**Comment by j4yn1ck5** [comment]
Same here. Windows 11 Zen b3. The single toolbar workaround isn't working for me. I already tried the Scroll Lock button. But my arrow keys work in other application. The problem is local to Zen.

**Comment by mr-cheffy** [comment]
Could anyone send me a link to a site that's *supposed* to work with arrow keys? I just cant find a single form that works with them, even in brave

**Comment by syscat29** [comment]
Can you try on a new profile? May sound crazy but some weird stuff has been going on lately with profiles and stuff that's supposed to be handled by the OS

**Comment by psv2522** [comment]
> Could anyone send me a link to a site that's _supposed_ to work with arrow keys? I just cant find a single form that works with them, even in brave

The problem with arrow keys is not just in the website but every input box, lets say if we paste something and one word is wrong we cannot move there with shift+arrow keys to correct it, we cannot accept the suggestions in urlbars, we cannot move forward or back in any video player(including youtube but you can do there with j or l as alternative) suggesting this does not affect anything is absurd

Edit: Temporary fix: If you want arrow keys to work downgrade to b1 and turn off auto update it works fine in that version

**Comment by MemQu** [comment]
> Could anyone send me a link to a site that's _supposed_ to work with arrow keys? I just cant find a single form that works with them, even in brave

Even scrolling the page doesn't work. I suggest trying any news site, using ublock-origin as a test case. You can even try commenting on this issue, as I've found out a few times!

(I'm sorry for the duplicate, but this is a major issue that wasn't catching attention. I've been primarying firefox as I click on extensions frequently)

**Comment by MemQu** [comment]
> Opening DevTools also triggers the bug

Do you still have this in 1.0.2-b.3? I'm not able to reproduce on windows 1.0.2-b.3.

**Comment by kuqquk** [comment]
Yes, I still have this issue on 1.0.2-b.3 on Windows. 
And I found this only happens when topbar is hidden in compact mode (with option "hide topbar" or "hide both"). 

To reproduce:
1. set topbar hidden in compact mode.
2. click any button of extension.
3. then arrow keys broken.

Then if you disable compact mode (<kbd> Ctrl + Alt + C</kbd>), and click any extension button and enable compact mode again, arrow keys are alive.

> > Opening DevTools also triggers the bug
> 
> Do you still have this in 1.0.2-b.3? I'm not able to reproduce on windows 1.0.2-b.3.

**Comment by j4yn1ck5** [comment]
> Then if you disable compact mode ( Ctrl + Alt + C), and click any extension button and enable compact mode again, arrow keys are alive.

This workaround doesn't work for me either. 

I finally restarted my computer. (a rare occurence) And that fixed it so far. I cannot reproduce any breakage and fixage coming from futzing around with extensions or display preference.

**Comment by B-Kingston** [comment]
Experiencing same issue (caused by clicking extensions), M1 Macbook Pro, macOS 15.2 in Zen 1.0.2-b.3.

**Comment by MemQu** [comment]
> Yes, I still have this issue on 1.0.2-b.3 on Windows. And I found this only happens when topbar is hidden in compact mode (with option "hide topbar" or "hide both").
> 
> To reproduce:
> 
>     1. set topbar hidden in compact mode.
> 
>     2. click any button of extension.
> 
>     3. then arrow keys broken.
> 
> 
> Then if you disable compact mode ( Ctrl + Alt + C), and click any extension button and enable compact mode again, arrow keys are alive.
> 
> > > Opening DevTools also triggers the bug
> > 
> > 
> > Do you still have this in 1.0.2-b.3? I'm not able to reproduce on windows 1.0.2-b.3.

I was referring to the specific case with dev tools (the quoted user saw). Only using extensions cause this as far as I can replicate on my side.

**Comment by j4yn1ck5** [comment]
Okay I was able to reproduce a fixing of the issue with the kuqquk method once. It seemed to require using a button from *within* an extension, not just clicking the extension itself. But, then I was unable to reproduce breaking the issue in the described manner. It seems for me like nothing is a 100% reproduce. And the breaking mechanism is still just a mysterious unobserved occurrence.

**Comment by mr-cheff** [comment]
Guys, are we sure this isn't a firefox issue or an user error? I dont know what everyone means with "arrow keys not working", but on the google search suggestions, I can move up and down with the arrows in zen so unless someone sends me a video showing how a site is not working with the arrows, ill just close this issue.

**Comment by mr-cheffy** [comment]
https://github.com/user-attachments/assets/94d5efa3-c6ef-4f03-aaff-a6c3e21c3f0a

**Comment by mr-cheffy** [comment]
Trying to fix it blindly haha, because I can't reproduce

**Comment by Ahmad-Mtr** [comment]
I had the same issue as others mentioned when I press on Extensions (especially Dark reader).
I'll try the new release and comment here afterwards.
Win 10.

**Comment by TheDavidFactor** [comment]
I saw an email asking for a video example from Mr m, don't see the comment here, but here is a video. Version:
![image](https://github.com/user-attachments/assets/c445b1eb-9d3a-4f83-b3b4-cdb48166048a)

https://github.com/user-attachments/assets/492f05fd-491d-4ab6-a553-c007d836c09d

**Comment by mr-cheffy** [comment]
Thanks a lot! Ill take a close look now. So it's apparently only with extension popups, right? I did take a glance over it and seems that when closing the popup, firefox registers arrow keys as `AltLeft` instead of `ArrowLeft`... super weird...

Also, is it only on compact mode?

**Comment by neurokitti** [comment]
this is a test to see if the arrow keys work on windows. they are working fine for me on twilight unless I am misunderstanding the bug.

https://github.com/user-attachments/assets/8acebde0-e60b-4fd1-8175-273d27d00fbd

**Comment by neurokitti** [comment]
ignore the close that was a misclick

**Comment by Ahmad-Mtr** [comment]
> Also, is it only on compact mode?

@mr-cheff yes, it only happens on compact mode when I interact with extensions.

**Comment by mr-cheffy** [comment]
The fix I tried to do in https://github.com/zen-browser/desktop/commit/dd88086cc7f3538e3d7d4edf9d69fb7c398ff771 has apparently "fixed" the issue according to @neurokitti. Ill release today and let's hope it works now 👍🏽. If it doesn't we can keep investigating into this matter

**Comment by EgoisticEscapist** [comment]
not working, still can't use arrow keys after clicking extensitons on compact mode in 1.0.2-b.4

https://github.com/user-attachments/assets/5598c4b6-b9ac-447d-b890-85dc7be49e6d

**Comment by TheDavidFactor** [comment]
I also continue to see the issue on Windows 11 on 1.0.2-b.4.
Additional troubleshooting info: turning off compact mode and then interacting with an extension restored normal arrow key function without having to restart Zen

**Comment by j4yn1ck5** [comment]
Windows 11 - 1.0.2-b.4. This build made the issue much more consistent and reproducible for me than it was before. 

I open Jason Savard's Checker Plus for Gmail and press the internal refresh button. Arrow keys stop working.
I hit ctrl-alt-c to disable compact mode, and hit Checker Plus internal refresh button. Arrow keys start working.
I hit ctrl- alt-c to enable compact mode. Repeat forever.

I also notice some issues with the extension displaying its user interface correctly sometimes, like it opens a much taller slab displaying itself several lines further down the screen or not at all.

**Comment by j4yn1ck5** [comment]
The issues I've noticed with extension UI displaying incorrectly are not limited to the one extension. It seems to affect many.

**Comment by j4yn1ck5** [comment]
> I also notice some issues with the extension displaying its user interface correctly sometimes, like it opens a much taller slab displaying itself several lines further down the screen or not at all.

> The issues I've noticed with extension UI displaying incorrectly are not limited to the one extension. It seems to affect many.

This incorrect extension UI display issue I mentioned looks fixed in 1.0.2-b5 as the changelog states.

Alas, the arrow keys bug is still reproducible.

**Comment by d0tlake** [comment]
The issue seems to be related to the act of clicking on the space outside of an extension to close its pop-up, and not just the act of interacting with an extension. I find that closing an extension by clicking on the button again will still allow me to use my arrow keys and other shortcuts, but the second I try closing an extension by clicking outside of the extension, then everything stops working.

An example: I have ublock origin pinned as a button on my toolbar. If I click the button to open the pop-up, and then press the button again to close it, arrow keys and shortcuts work. If I click on the button, then click on any space in my browser to close the pop-up, then the arrow keys and shortcuts stop working.

**Comment by j4yn1ck5** [comment]
I just tested d0tlake's reproduce a whole bunch. And it is substantiated on my end as well.

I have a couple interesting notes to add. With this reproduction method, I don't need to disable compact mode to fix the problem. What I notice is that if I click outside the extension pop-up as described, the arrow keys break. And if I then click the same extension and dismiss it by clicking the same extension's icon, the arrow keys are then fixed. However, if I break the arrow keys by outside dismissing an extension, I can *only* fix the broken arrows keys with that same extension. Trying to dismiss via extension icon on any other extension will not fix it. It must be the same one that broke it.

This reproduction method does not seem to be limited to any specific extension. It seems to work on all of them, provided that I use the same extension to fix the problem after it's broken.

**Comment by sergioabreu-g** [comment]
Same here on Pop!_OS 22.04 LTS with Zen version 1.0.2-b.5 (Firefox 133.0.3) (64-bit).

@j4yn1ck5 reproduction steps are spot on. It happens to me only when in compact mode with the top bar hidden. Clicking outside of an extension to close its popup triggers it, while closing that same extension by pressing again on its button fixes it (or prevents it from happening in the first place). Enabling/disabling compact mode while the bug is happening doesn't fix it for me, only closing the extension popup with its own button does.

**Comment by mr-cheffy** [comment]
Funny how it doesn't happen on the dev builds...

**Comment by mr-cheffy** [comment]
Right, so why is it only on compact mode...

Non compact mode:

![Image](https://github.com/user-attachments/assets/c5d1f3ce-da4d-4d30-9a23-6113cb800804)
 

Compact mode:

![Image](https://github.com/user-attachments/assets/4cd1b976-b2d0-4ab5-9556-c3a05f7a7d5c)

**Comment by mr-cheffy** [comment]
Ok, im getting closer to the issue...

The compact mode toolbar hides when there's no `open=true` popups. When closing this popup, that toolbar closes/hides before the popup finishes the animation, leaving it confused as in whether or not it is closed or opened... 

edit: can someone try out twilight build (https://github.com/zen-browser/desktop/actions/runs/12530803176) once it finishes building? See if it's fixed (https://github.com/zen-browser/desktop/commit/7ea0630973a602e031ccf97df53e1319fc8bfe49), that would be very helpful...

**Comment by jonaskohl** [comment]
Good catch! The issue seems to be resolved in that Twilight build! Just tried it on macOS 15.2 and had no issues so far (while the issue in the stable release 1.0.2-b.5 is still there)

**Comment by mr-cheffy** [comment]
Great! 🥳

**Comment by teeaggo** [comment]
Same issue here. Using version Zen 1.0.2-b.5 on Windows 10 Pro for Workstations, build 19045.5247. Arrow keys and navigation keys: (Home, End)

**Comment by mr-cheffy** [comment]
> Same issue here. Using version Zen 1.0.2-b.5 on Windows 10 Pro for Workstations, build 19045.5247. Arrow keys and navigation keys: (Home, End)

Yeah, it'll be fixed on b.6 👍🏽

**Comment by mr-cheffy** [comment]
I've did a slight change on how we handle compact mode. Could anyone try it out on the latest twilight? Thanks

**Comment by GJRobert** [comment]
> I've did a slight change on how we handle compact mode. Could anyone try it out on the latest twilight? Thanks

Thank you @mr-cheff I just tried on Twilight 1.6t and clicking outside extension popups does not break the arrow keys, tab key etc.

Compact Mode is crucial / the core for my usage of Zen Browser and thanks for everyone in this thread!

**Comment by GJRobert** [comment]
By the way, this also fixes another bug: **extension buttons put on the bottom toolbar of Compact Mode side-tabbar, popups are opened but will keep open even if mouse clicking outside the popup** (including the profile switcher button of Zen). Now I can put the extension buttons back there, hurray!

**Comment by BoscoDomingo** [comment]
This is still an issue as of today (2025/01/08).

Edit: Just saw it's fixed on Twilight. Since it hasn't been published to the Beta, should it really be closed? It is quite useful to keep issues open until they are truly released IMO

**Comment by fangmarks** [comment]
> Windows 11 - 1.0.2-b.4. This build made the issue much more consistent and reproducible for me than it was before.
> 
> I open Jason Savard's Checker Plus for Gmail and press the internal refresh button. Arrow keys stop working. I hit ctrl-alt-c to disable compact mode, and hit Checker Plus internal refresh button. Arrow keys start working. I hit ctrl- alt-c to enable compact mode. Repeat forever.
> 
> I also notice some issues with the extension displaying its user interface correctly sometimes, like it opens a much taller slab displaying itself several lines further down the screen or not at all.

Thank you, doing that also brought it back for me just now



> This is still an issue as of today (2025/01/08).
> 
> Edit: Just saw it's fixed on Twilight. Since it hasn't been published to the Beta, should it really be closed? It is quite useful to keep issues open until they are truly released IMO

Good to know it's fixed there! And yeah, I agree, the Issue should ideally only be fully closed once it gets merged into Stable

**Comment by TheDavidFactor** [comment]
Just wanted to update and let you know that I just updated 1.6b and the issue appears to be fixed for me
